stdlib/public/runtime/Demangle.cpp

```diff
@@ -443,6 +444,53 @@ swift::_swift_buildDemanglingForMetadata(const Metadata *type,
     // Just a simple composition of protocols.
     return proto_list;
   }
+  case MetadataKind::ExtendedExistential: {
+    auto exis = static_cast<const ExtendedExistentialTypeMetadata *>(type);
+    auto genSig = exis->Shape->getGeneralizationSignature();
+    const unsigned selfParamIdx = genSig.getParams().size();
+    auto node = Dem.createNode(Node::Kind::ParameterizedProtocol);
+    for (const auto &reqt :
+         exis->Shape->getRequirementSignature().getRequirements()) {
+      if (reqt.getKind() != GenericRequirementKind::Protocol) {
+        continue;
+      }
+
+      if (!reqt.Flags.hasKeyArgument()) {
+        continue;
+      }
+
+      auto lhsTypeNode = Dem.demangleType(reqt.getParam());
+      if (!lhsTypeNode || lhsTypeNode->getKind() != Node::Kind::Type ||
+          !lhsTypeNode->hasChildren() ||
+          lhsTypeNode->getChild(0)->getKind() !=
+              Node::Kind::DependentGenericParamType ||
+          lhsTypeNode->getChild(0)->getNumChildren() != 2) {
+        continue;
+      }
+      auto index = lhsTypeNode->getChild(0)->getChild(1)->getIndex();
+      if (index + 1 != selfParamIdx)
+        continue;
+
+      auto *protocolNode =
+          _buildDemanglingForProtocolDescriptor(reqt.getProtocol(), Dem);
+      if (!protocolNode)
+        continue;
+
+      node->addChild(protocolNode, Dem);
+    }
+
+    const unsigned shapeArgumentCount =
+        exis->Shape->getGenSigArgumentLayoutSizeInWords();
+    auto type_list = Dem.createNode(Node::Kind::TypeList);
+    for (unsigned i = 0; i < shapeArgumentCount; ++i) {
+      auto genArg = exis->getGeneralizationArguments()[i];
+      auto eltType =
+          _swift_buildDemanglingForMetadata((const Metadata *)genArg, Dem);
+      type_list->addChild(eltType, Dem);
+    }
```

Comment:
Hmm.  Technically this isn't correct; there's no reason the generalization arguments have to be 1-1 with the primary associated types.  For example, it would break if we had a protocol composition with a class constraint.

Fortunately, the shape actually has a mangled type string which refers to the generalization arguments; we just need to demangle that under the generalization substitutions.  We should be able to leverage existing functions to do that easily enough.